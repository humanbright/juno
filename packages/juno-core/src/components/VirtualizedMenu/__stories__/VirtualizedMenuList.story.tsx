import React, { ComponentProps } from 'react';

import range from 'lodash/range';

import {
  RcListItemText,
  RcMenuItem,
  RcVirtualizedMenuList,
} from '@ringcentral/juno';
import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
} from '@ringcentral/juno-storybook';
import { Meta, Story } from '@storybook/react';

export default {
  title: 'ðŸš€ Cleanup Components/VirtualizedMenu/VirtualizedMenuList',
  component: RcVirtualizedMenuList,
  argTypes: {
    ...sortInDocTable<keyof MenuProps>([]),
    ...notControlInDocTable<keyof MenuProps>([]),
    ...notShowInDocTable<keyof MenuProps>([]),
  },
} as Meta;

type MenuProps = ComponentProps<typeof RcVirtualizedMenuList>;

const menus = range(0, 1000);

export const VirtualizedMenuList: Story<Partial<MenuProps>> = ({
  children,
  ...args
}) => {
  return (
    <RcVirtualizedMenuList {...args}>
      {menus.map((x) => (
        <RcMenuItem key={x}>
          <RcListItemText primary={`MenuItem ${x}`} />
        </RcMenuItem>
      ))}
    </RcVirtualizedMenuList>
  );
};

VirtualizedMenuList.args = {
  maxHeight: 180,
  VirtuosoProps: {
    // view more in https://virtuoso.dev/virtuoso-api-reference/
    overscan: 200,
  },
};

VirtualizedMenuList.argTypes = {
  ...notControlInDocTable<keyof MenuProps>([]),
};

VirtualizedMenuList.parameters = {
  tags: [
    {
      name: 'Mui',
      href: 'https://material-ui.com/api/menu-list/',
      value: 'MenuList',
    },
    {
      name: 'Source',
      value: 'virtuoso',
      href: 'https://virtuoso.dev/',
      color: 'avatar.lake',
    },
    {
      name: 'Accessibility',
      value: '100%',
    },
  ],
};

VirtualizedMenuList.storyName = 'VirtualizedMenuList';
