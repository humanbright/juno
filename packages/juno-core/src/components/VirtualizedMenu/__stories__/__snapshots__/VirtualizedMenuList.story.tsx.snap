// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/VirtualizedMenu/VirtualizedMenuList VirtualizedMenuList 1`] = `
Array [
  .c0 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<span
    className="c0"
    onFocus={[Function]}
  />,
  .c0 {
  height: 8px;
}

<div
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "WebkitOverflowScrolling": "touch",
        "height": 180,
        "outline": "none",
        "overflowY": "auto",
        "position": "relative",
      }
    }
    tabIndex={-1}
  >
    <div
      style={
        Object {
          "height": "100%",
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    >
      <div>
        <div
          className="c0"
        />
      </div>
      <ul
        className="MuiList-root "
        role="menu"
        style={
          Object {
            "boxSizing": "border-box",
            "marginTop": 0,
            "paddingBottom": 0,
            "paddingTop": 0,
          }
        }
      />
      <div>
        <div
          className="c0"
        />
      </div>
    </div>
  </div>,
]
`;
